userspace/falco/stats_writer.cpp

```diff
@@ -246,7 +246,7 @@ void stats_writer::worker() noexcept {
 
 		// this helps waiting for the first tick
 		tick = stats_writer::get_ticker();
-		if(first_tick != tick) {
+		if(m_first_run || first_tick != tick) {
```

Comment:
Can't be `static` since we have to account for the hot reload conditions.

Reply:
I am wondering whether it would be easier to instead declare eg:
```
auto first_tick = -1;
```
so that
```
tick = stats_writer::get_ticker();
if(first_tick != tick) {
```
is always true on first iteration.
This way, we could avoid the `m_first_run` bool entirely. I double checked, i don't see `first_tick` used elsewhere, i think it was meant for the purpose of skipping the very first tick.
Perhaps we can get rid of `first_tick` entirely too.

Reply:
Overlooked this earlier today @FedeDP - I just re-pushed the co-authored cleanup commit. Yes I agree this is more clever and as far as I can tell it works just fine! Thanks for the suggestion!